### PR TITLE
Standardize data types

### LINKED ISSUE #6: Proposal: Replace `usize` with `uN`
`occurrence` and `line_count` variables shouldn't be limited by the users' system architecture, therefore allowing the maximum possible value `u128`.

### PR DESCRIPTION
**Pull request by yoimon**
## Summary
- Replaced all occurrences of `usize` with `u128`.
- Converted *occurrence* to `u128`
- Bumped version to **0.3.0**

Resolves #6

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -113,12 +113,12 @@ pub fn run(config: Config) -> Result<(), Box<dyn Error>> {
...
         println!("No match found for {} in {}", &config.pattern, &config.file);
     } else {
-        let (mut line_count, mut total_occurrence): (usize, usize) = (0usize, 0usize);
+        let (mut line_count, mut total_occurrence): (u128, u128) = (0u128, 0u128);
 
         let mut matched_result: String = String::new();
 
         for line in result {
-            let occurrence: usize = line.matches(&config.pattern).count();
+            let occurrence: u128 = line.matches(&config.pattern).count().try_into().unwrap();
```
Comment (m0lese) [inline src/lib.rs:121]: I'm not sure whether converting it to `u128` is a good idea. Have to research. I'll let it roll for now.

**Review by m0lese — APPROVED** [review]
LGTM!

File: src/lib.rs (hunk 1 above)
Comment (yoimon) [reply src/lib.rs:121]: Great!

File: src/lib.rs [hunk 2]
```diff
@@ -113,12 +113,12 @@ pub fn run(config: Config) -> Result<(), Box<dyn Error>> {
...
         println!("No match found for {} in {}", &config.pattern, &config.file);
     } else {
-        let (mut line_count, mut total_occurrence): (usize, usize) = (0usize, 0usize);
+        let (mut line_count, mut total_occurrence): (u128, u128) = (0u128, 0u128);
 
         let mut matched_result: String = String::new();
 
         for line in result {
-            let occurrence: usize = line.matches(&config.pattern).count();
+            let occurrence: u128 = line.matches(&config.pattern).count();
```
Comment (m0lese) [inline src/lib.rs]:
```suggestion
let occurrence: u128 = line.matches(&config.pattern).count().try_into().unwrap();
```

**Pull request merged**